front/pages/api/w/[wId]/data_sources/request_access.ts

```diff
@@ -0,0 +1,136 @@
+import type { Result } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import { isLeft } from "fp-ts/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import config from "@app/lib/api/config";
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { sendEmail } from "@app/lib/email";
+import logger from "@app/logger/logger";
+import { apiError } from "@app/logger/withlogging";
+
+export const PostRequestAccessBodySchema = t.type({
+  emailMessage: t.string,
+  email: t.string,
```

Comment:
Reading this one more time, we should make sure to include the user's sId here instead of email. This way, we can prevent any misuse of the endpoint, like sending emails to random addresses. Just a little extra precaution to keep things secure!